__tests__/cfn-integration-test.js

```diff
@@ -2,111 +2,111 @@ const fs = require('fs')
-
-        // pin versions of SDKs
-        const cfn = new AWS.CloudFormation({ region: region }),
-              s3 = new AWS.S3({ region: region }),
-              logger = function (input) {
-                  _console(`${region}:${stackName}:${input}`)
-              }
-
-        // Upload the packaged template to S3, then use the resulting URL in the CFN createStack call
-        // This is necessary because the file is too large to deliver in-line to CFN
-        cfnParams.TemplateURL =(await s3.upload(s3Params).promise()).Location
-
-        logger('createStack call starting.')
-        await cfn.createStack(cfnParams).promise()
-        logger('createStack call succeeded.')
-
-        logger('stackExists waiter starting.')
-        await cfn.waitFor('stackExists', { StackName: stackName }).promise()
-        logger('stackExists waiter succeeded.')
-
-        logger('stackCreateComplete waiter starting.')
-        let devPortalUrl =
-            (await cfn.waitFor('stackCreateComplete', { StackName: stackName }).promise()).Stacks[0].Outputs
-                .find((output) => output.OutputKey === 'WebsiteURL').OutputValue
-        logger('stackCreateComplete waiter succeeded.')
+  // NOTE: These tests all assume that the CFN template has already been packaged *PER REGION*!
+  const cfnTimeout = 75
+  // run the test with a timeout of slightly longer than double the CFN stack timeout
+  const testTimeout = 1000 * 60 * cfnTimeout * 2 + 1
+  const _console = console.log
+
+  async function commonTest (region, stackMiddlefix) {
+    const unixTimestamp = Math.floor(new Date() / 1000)
+    const stackName = `cfn-integ-${stackMiddlefix}-${unixTimestamp}`
+    const s3Params = {
+      // CFN, when reading the template from S3, requires the S3 bucket to be in the same region as the CFN stack...
+      Bucket: `dev-portal-integ-${region}`,
+      Body: fs.readFileSync(`./cloudformation/packaged-${region}.yaml`),
+      Key: stackName
+    }
+    const cfnParams = {
+      StackName: stackName,
+      Capabilities: ['CAPABILITY_NAMED_IAM', 'CAPABILITY_AUTO_EXPAND'],
+      Parameters: [
+        {
+          ParameterKey: 'DevPortalSiteS3BucketName',
+          ParameterValue: `integ-${stackMiddlefix}-${unixTimestamp}-dev-portal-test`
+        },
+        {
+          ParameterKey: 'ArtifactsS3BucketName',
+          ParameterValue: `integ-${stackMiddlefix}-${unixTimestamp}-artifact-bucket`
+        },
+        {
+          ParameterKey: 'DevPortalCustomersTableName',
+          ParameterValue: `Customers${unixTimestamp}`
+        },
+        {
+          ParameterKey: 'CognitoDomainNameOrPrefix',
+          ParameterValue: `integ-${stackMiddlefix}-${unixTimestamp}`
+        }
+      ],
+      // RoleARN: 'STRING_VALUE',
+      TimeoutInMinutes: cfnTimeout
+    }
 
-        logger(`verifying that stack is available at ${devPortalUrl} .`)
-        let staticIndex = await rp(devPortalUrl)
+    console.log('commonTest', region)
```

Comment:
I see `console.log`, `_console` and `logger` - do we need all three?

Reply:
Probably not, but we can revisit it later.